cedar-policy/src/ffi/is_authorized.rs

```diff
@@ -443,12 +443,50 @@ pub enum PartialAuthorizationAnswer {
 pub struct AuthorizationCall {
     /// The principal taking action
     #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
-    principal: Option<JsonValueWithNoDuplicateKeys>,
+    principal: JsonValueWithNoDuplicateKeys,
     /// The action the principal is taking
     #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
     action: JsonValueWithNoDuplicateKeys,
     /// The resource being acted on by the principal
     #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
+    resource: JsonValueWithNoDuplicateKeys,
+    /// The context details specific to the request
+    #[serde_as(as = "MapPreventDuplicates<_, _>")]
+    #[cfg_attr(feature = "wasm", tsify(type = "Record<string, CedarValueJson>"))]
+    /// The context details specific to the request
+    context: HashMap<String, JsonValueWithNoDuplicateKeys>,
+    /// Optional schema.
+    /// If present, this will inform the parsing: for instance, it will allow
+    /// `__entity` and `__extn` escapes to be implicit, and it will error if
+    /// attributes have the wrong types (e.g., string instead of integer).
+    #[cfg_attr(feature = "wasm", tsify(optional, type = "Schema"))]
+    schema: Option<Schema>,
+    /// If this is `true` and a schema is provided, perform request validation.
+    /// If this is `false`, the schema will only be used for schema-based
+    /// parsing of `context`, and not for request validation.
+    /// If a schema is not provided, this option has no effect.
+    #[serde(default = "constant_true")]
+    validate_request: bool,
+    /// The slice containing entities and policies
+    slice: RecvdSlice,
+}
+
+/// Struct containing the input data for partial authorization
+#[cfg(feature = "partial-eval")]
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "wasm", derive(tsify::Tsify))]
+#[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
+#[serde(rename_all = "camelCase")]
+pub struct PartialAuthorizationCall {
+    /// The principal taking action
+    #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
+    principal: Option<JsonValueWithNoDuplicateKeys>,
+    /// The action the principal is taking
+    #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
+    action: Option<JsonValueWithNoDuplicateKeys>,
+    /// The resource being acted on by the principal
+    #[cfg_attr(feature = "wasm", tsify(type = "{type: string, id: string}"))]
```

Comment:
the TSify types should be updated; I think the correct signature would be `null | {type: ...`